common/config/retryer.go

```diff
@@ -0,0 +1,107 @@
+package config
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+type Retryer struct {
+	c            client.DefaultRetryer
+	minRetryTime int
+	maxRetryTime int
+}
+
+func NewRetryer(minRetryTime, maxRetryTime int) *Retryer {
+	// minRetryTime default is 500ms
+	// maxRetryTime default is 5 minutes
+	retryer := &Retryer{
+		minRetryTime: 500,
+		maxRetryTime: 300000,
+	}
+
+	if minRetryTime > 0 {
+		retryer.minRetryTime = minRetryTime
+	}
+
+	if maxRetryTime > 0 {
+		retryer.maxRetryTime = maxRetryTime
+	}
+
+	return retryer
+}
+
+var seededRand = rand.New(&lockedSource{src: rand.NewSource(time.Now().UnixNano())})
+
+// RetryRules returns the delay duration before retrying this request again
+func (d Retryer) RetryRules(r *request.Request) time.Duration {
+	minTime := 30
+	throttle := d.shouldThrottle(r)
+	if throttle {
+		minTime = 500
+	}
+
+	retryCount := r.RetryCount
+	if retryCount > 13 {
+		retryCount = 13
+	} else if throttle && retryCount > 8 {
+		retryCount = 8
+	}
+
+	delay := (1 << uint(retryCount)) * (seededRand.Intn(minTime) + minTime)
```

Comment:
one reason I can see for not wanting this code implemented on our side is testing it :trollface:

Reply:
As stated above, this doesn't fix the root of problem we are trying to solve; it should be removed regardless.